### PR TITLE
API-657 GH actions

### PR DESCRIPTION
**Pull request by srknzl**
adds a pr builder and branch builder. we need to add HAZELCAST_ENTERPRISE_KEY secret to the repo

### CODE REVIEW

**Comment by srknzl** [comment]
verify

File: .github/workflows/test_branch.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Test Branch
```
Comment (mdumandag) [inline .github/workflows/test_branch.yml:1]: Can we merge these two files into one? They are almost exact copies of each other.
I think we can add another `on` rule below for PRs and pass `HAZELCAST_ENTERPRISE_KEY` conditionally as an environment variable. WDTY?

File: .github/workflows/test_branch.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        steps:
+            - name: Setup Java
+              uses: actions/setup-java@v1
+              with:
+                  java-version: 8
+            - name: Setup Node.js
+              uses: actions/setup-node@v2
+              with:
+                  node-version: 10
+            - name: Disable line ending conversion of git
```
Comment (mdumandag) [inline .github/workflows/test_branch.yml:27]: Why is this necessary?

Comment (srknzl) [reply .github/workflows/test_branch.yml:27]: On windows, git converts line endings from linux format to windows format. Eslint was printing a lot of warnings since we have a rule for line endings.

File: .github/workflows/test_branch.yml (hunk 1 above)
Comment (srknzl) [reply .github/workflows/test_branch.yml:1]: Makes sense, let me try

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+name: Test Branch
```
Comment (mdumandag) [inline .github/workflows/ci.yml:1]:
```suggestion
name: Run tests
```
WDYT? Test branch seems a little bit off to me, after the recent changes

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+on:
+    push:
+        branches-ignore:
+            - 'gh-pages'
+        tags-ignore:
+            - '*'
+    pull_request:
+
+jobs:
+    test-branch:
```
Comment (mdumandag) [inline .github/workflows/ci.yml:11]:
```suggestion
    run-tests:
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        name: Test Branch (${{ matrix.os }})
```
Comment (mdumandag) [inline .github/workflows/ci.yml:12]:
```suggestion
        name: Run tests on (${{ matrix.os }})
```

**Review by mdumandag — APPROVED** [review]
Looks good overall. I left some naming-related comments.

BTW, we should work on shortening the test duration as it is close to 30 min mark now. It is almost as long as the IMDG itself. We can maybe make use of parallel test runs or work on the long tests to make them simpler/shorter

**Comment by srknzl** [comment]
I checked mocha parallel test feature but I remember it was hard to enable because running tests in parallel affect other tests. I will try to do again. Opening a issue.

**Comment by srknzl** [comment]
After this is merged, should we ditch Jenkins @mdumandag ?
